### ISSUE TITLE
Incorrect tokenizer loaded from checkpoint

### ISSUE THREAD

**Issue by kspieks** [rxn4chemistry/rxn_yields#11]
Hello! I’m trying to reproduce the first row from Table 1 from your 2021 [paper](https://iopscience.iop.org/article/10.1088/2632-2153/abc81d). However, I’m encountering an error when trying to load the published model. Based on the trace back, I’m wondering if I’m reading from the correct .bin file? Can you please point me in the right direction?

```
The tokenizer class you load from this checkpoint is not the same type as the class this function is called from. It may result in unexpected tokenization. 
The tokenizer class you load from this checkpoint is 'BertTokenizer'. 
The class this function is called from is 'SmilesTokenizer'.
```

A minimal working example is below which mostly copies the evaluation code from your [website](https://rxn4chemistry.github.io/rxn_yields/results_evaluation_of_buchwald_hartwig_yields_prediction/) to just test predictions for one model. 

```
import logging
import os

import sklearn
import torch
from sklearn.metrics import mean_squared_error, r2_score, mean_absolute_error
import pandas as pd
from rxnfp.models import SmilesClassificationModel
from rxn_yields.data import generate_buchwald_hartwig_rxns
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches
import numpy as np

print(f'torch.cuda.is_available(): {torch.cuda.is_available()}')

try:
    import wandb
    wandb_available = True
except ImportError:
    wandb_available = False
from dotenv import load_dotenv, find_dotenv
load_dotenv(find_dotenv())

NAME_SPLIT = [
    ('FullCV_01', 2768),
    # ('FullCV_02', 2768),
    # ('FullCV_03', 2768),
    # ('FullCV_04', 2768),
    # ('FullCV_05', 2768),
    # ('FullCV_06', 2768),
    # ('FullCV_07', 2768),
    # ('FullCV_08', 2768),
    # ('FullCV_09', 2768),
    # ('FullCV_10', 2768),
    # ('Test1', 3058),
    # ('Test2', 3056),
    # ('Test3', 3059),
    # ('Test4', 3056),
]

def make_plot(y_test, y_pred, rsme, r2_score, mae, name):
    fontsize = 16
    fig, ax = plt.subplots(figsize=(8,8))
    r2_patch = mpatches.Patch(label="R2 = {:.3f}".format(r2_score), color="#5402A3")
    rmse_patch = mpatches.Patch(label="RMSE = {:.1f}".format(rmse), color="#5402A3")
    mae_patch = mpatches.Patch(label="MAE = {:.1f}".format(mae), color="#5402A3")
    plt.xlim(-5,105)
    plt.ylim(-5,105)
    plt.scatter(y_pred, y_test, alpha=0.2, color="#5402A3")
    plt.plot(np.arange(100), np.arange(100), ls="--", c=".3")
    plt.legend(handles=[r2_patch, rmse_patch, mae_patch], fontsize=fontsize)
    ax.set_ylabel('Measured', fontsize=fontsize)
    ax.set_xlabel('Predicted', fontsize=fontsize)
    ax.set_title(name, fontsize=fontsize)
    return fig

y_predictions = []
y_tests = []
r2_scores = []
rmse_scores = []

RXN_YIELDS_PATH = os.path.join('/home', 'kspieker', 'code', 'rxn_yields')

for (name, split) in NAME_SPLIT:
    df_doyle = pd.read_excel(os.path.join(RXN_YIELDS_PATH, 
    					'data', 
    					'Buchwald-Hartwig',
    					'Dreher_and_Doyle_input_data.xlsx',
    					),
    					sheet_name=name,
    					engine='openpyxl',
    					)

    df_doyle['rxn'] = generate_buchwald_hartwig_rxns(df_doyle)

    train_df = df_doyle.iloc[:split-1][['rxn', 'Output']] # paper has starting index 1 not 0
    test_df = df_doyle.iloc[split-1:][['rxn', 'Output']]  # paper has starting index 1 not 0

    train_df.columns = ['text', 'labels']
    test_df.columns = ['text', 'labels']
    
    y_test = test_df.labels.values

    mean = train_df.labels.mean()
    std = train_df.labels.std()

    model_type='bert'
    model_path = os.path.join(RXN_YIELDS_PATH,
                              'trained_models',
                              'buchwald_hartwig',
                               f"{name}_split_{str(split).replace('-','_')}",
                              'checkpoint-1730-epoch-10',
                              )
    print(f'model_path: {model_path}')
    model = SmilesClassificationModel(model_type, model_path,
                                      num_labels=1, args={
                                      "regression": True
                                      }, 
                                      use_cuda=torch.cuda.is_available(),
                                      )
    
    y_preds = model.predict(test_df.text.values)[0]
    
    y_preds = y_preds * std + mean

    r_squared = r2_score(y_test, y_preds)
    rmse = mean_squared_error(y_test, y_preds) ** 0.5
    mae = mean_absolute_error(y_test, y_preds)
    r2_scores.append(r_squared)
    
    print(f"{name} | R2 {r_squared:.2f} | RMSE {rmse:.1f} | MAE {mae:.1f}")
    fig = make_plot(y_test, y_preds, rmse, r_squared, mae, name)
        
    fig.savefig(f'images/mlst_doyle_results/{name}_{split}.pdf')

    # now clip the predictions like they do in the script
    # https://rxn4chemistry.github.io/rxn_yields/results_evaluation_of_buchwald_hartwig_yields_prediction/
    print('Clipping predictions...')
    y_preds = np.clip(y_preds, 0, 100)

    r_squared = r2_score(y_test, y_preds)
    rmse = mean_squared_error(y_test, y_preds) ** 0.5
    mae = mean_absolute_error(y_test, y_preds)
    r2_scores.append(r_squared)
    
    print(f"{name} | R2 {r_squared:.2f} | RMSE {rmse:.1f} | MAE {mae:.1f}")
    fig = make_plot(y_test, y_preds, rmse, r_squared, mae, name)
        
    fig.savefig(f'images/mlst_doyle_results/{name}_{split}_clipped.pdf')
```

As a separate question, can you also post the SI? I’m having trouble accessing it as the link from page 2 of the main text seems to be expired. 

Thanks in advance for your help!

**Comment by kspieks** [comment]
@pschwllr are you able to help answer these questions?

**Comment by pschwllr** [comment]
Are you sure that you are using `transformers==2.11.0 torch==1.5.1 simpletransformers==0.34.4`?

The SI is the following:
[si.pdf](https://github.com/rxn4chemistry/rxn_yields/files/8897915/si.pdf)

Please note that there was a bug with the dropout in `simpletransformers==0.34.4` (it was deactivated after one epoch, hence it had not much of an influence on the performance of the models). 

In a new project, I would directly use [Huggingface Transformers](https://github.com/huggingface/transformers) and redo the hyperparameter optimisation for the dropout.

**Comment by kspieks** [comment]
Thanks your reply. I was accidentally using a newer version of `simpletransformers`. After a fresh install, the script above loads the published model.

Thanks for sharing the SI and giving a heads up about the dropout bug.
